### DISCUSSION TITLE
Extend control over middlewares in `middlewares.init.before` hook

### DISCUSSION THREAD

**Discussion by viters** [directus/directus#8713] [Feature Requests]
Right now it is impossible to add another middleware in between existing ones (it is possible only at the start or at the end of declared middlewares in `app.ts`). Due to certain middlewares implementation, it might be sometimes hard or even impossible for custom middlewares to run, without applying them at the specific index. 

Moreover, it should also be possible to override Directus middlewares, reorder them or remove them (it would be a feature to advanced users).

My idea is to push middlewares to `middlewares` array and provide it through `middlewares.init.before`, so user can modify it, before passing it to `app.use`.

Proposal can be implemented without breaking changes (just by extending current API of `middlewares.init.before`) and with a little refactor to how middlewares are declared in `app.ts`. I will send a PR in a moment.

**Comment by rijkvanzanten** [comment]
> Moreover, it should also be possible to override Directus middlewares, reorder them or remove them (it would be a feature to advanced users).

I don't necessarily agree with this. Messing with the system internals at that level can cause all sorts of unexpected errors, which we should really prevent from happening. For any particular overrides or configuration that's required in the existing middleware, we should look into appropriate configuration env vars or init hooks.

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
